client/src/components/DataFiles/DataFilesToolbar/DataFilesToolbar.js

```diff
@@ -190,7 +199,7 @@ const DataFilesToolbar = ({ scheme, api }) => {
           text="Download"
           iconName="download"
           onClick={download}
-          disabled={!canDownload}
+          disabled={!canDownload && !canCompress}
```

Comment:
I'm curing about the `&& !canCompress` change.  So for selecting multiple files (like if a user selects two text files), the Download button isn't disabled and the user would be shown saying _Folders must be compressed before downloaded_ which would be confusing as they haven't even selected a file. 🤔 

The compress file permissions (i.e `getFilePermissions('compress'`) also check

Reply:
This logic does cause the problem @nathanfranklin mentions: [(screenshot) Two Files Causes Folder Download Message](https://user-images.githubusercontent.com/62723358/122063517-f51c9d80-cdb5-11eb-8611-ad003a9294e1.png).

__What is the intent of this logic?__
_My guess: If user has selected a folder and folder cannot be compressed, then disable button because user can **not** download **nor** compress folder._

__The error seems to come from `case 'download'` at [utils/filePermissions.js](https://github.com/TACC/Core-Portal/blob/main/client/src/utils/filePermissions.js#L31-L36).__
_Maybe, the logic for that case should instead be (verify my logic, please):_
```
        (
          ( files.length === 1 && files[0].format !== 'folder' ) ||
          files[0].format === 'folder'
        ) &&
        api !== 'googledrive'
```

Reply:
I can reproduce that scenario, let me know if this is correct:
- The user only can download one file at a time but can compress multiple files/folders

If that's correct the Download button will be:
- Enabled when at least one folder is selected (with any number of files or no file) **or** only one file is selected
- Disabled when multiple files are selected **and** no folder is selected

Please let me know if that logic is correct. One suggestion is to change the dialog wording to let the user know that they are allowed to download one file at a time or they should compress and download.

Reply:
I think that makes sense to me @jmartinez-tacc. You could take it a step further and always have the button enabled, but show the modal if either a folder is selected, or multiple objects are selected (files or folders). I also like your suggestion. Perhaps the message should also include something like "To download multiple files, you must first compress them like you would a folder".

Thoughts @tacc-wbomar ?

Reply:
> This logic does cause the problem @nathanfranklin mentions: [(screenshot) Two Files Causes Folder Download Message](https://user-images.githubusercontent.com/62723358/122063517-f51c9d80-cdb5-11eb-8611-ad003a9294e1.png).
> 
> What is the intent of this logic?
> My guess: If user has selected a folder and folder cannot be compressed, then disable button because user can **not** download **nor** compress folder.
> 
> The error seems to come from `case 'download'` at [utils/filePermissions.js](https://github.com/TACC/Core-Portal/blob/main/client/src/utils/filePermissions.js#L31-L36).
> Maybe, the logic for that case should instead be (verify my logic, please):
> 
> ```
>         (
>           ( files.length === 1 && files[0].format !== 'folder' ) ||
>           files[0].format === 'folder'
>         ) &&
>         api !== 'googledrive'
> ```

The logic that causes the problem is the 'compress' case, due two files can compress it enables the download button, but dialog message is intended for folders only.

Reply:
> I can reproduce that scenario, let me know if this is correct:
> 
> * The user only can download one file at a time but can compress multiple files/folders
> 
> If that's correct the Download button will be:
> 
> * Enabled when at least one folder is selected (with any number of files or no file) **or** only one file is selected
> * Disabled when multiple files are selected **and** no folder is selected
> 
> Please let me know if that logic is correct. One suggestion is to change the dialog wording to let the user know that they are allowed to download one file at a time or they should compress and download.

For now I'm gonna stick with the original business logic. If there's change let me know or we can create another task.

Reply:
> Thoughts @tacc-wbomar?

I think we follow the original business logic.

I believe this means remove the ` && !canCompress`.

And if there is a UX question, ask in [#cep-design](https://tacc-team.slack.com/archives/CQUA4D5KJ).

Reply:
P.S. Design was also thinking about this independently and has a [solution](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1624033021085600).